src/urldecode.cpp

```diff
@@ -32,3 +32,23 @@ CharacterVector url_decode2(CharacterVector url) {
     return output;
   });
 }
+
+std::string str_reverse(std::string x) {
+  std::reverse(x.begin(), x.end());
+  return x;
+}
+
+//[[Rcpp::export]]
+CharacterVector url_reverse(CharacterVector urls) {
+  unsigned int input_size = urls.size();
+  CharacterVector output(input_size);
+  for (unsigned int i = 0; i < input_size; i++) {
+    if (urls[i] == NA_STRING) {
```

Comment:
If you want to be terse:

```cpp
// untested but should work.
output[i] = urls[i] == NA_STRING ? NA_STRING : str_reverse(Rcpp::as<std::string>(urls[i]));
```

Reply:
But that's how people write unreadable C code.

Reply:
I think i keep it as is for future me ;-)

Reply:
Checked:

```cpp
    output[i] = (urls[i] == NA_STRING) ? NA_STRING : String(str_reverse(Rcpp::as<std::string>(urls[i])));
```

But don't use.